### PR TITLE
Add shadowing warning flag, and fix shadowing bugs

### PR DESCRIPTION
**Pull request by n-hutton**
Fixes found when turning on -Wshadow - also force naming convention for fetch tests

### CODE REVIEW

File: libs/core/include/core/logger.hpp [hunk 1]
```diff
@@ -787,6 +787,31 @@ extern log::details::LogWrapper logger;
 
 #endif
 
+#define ERROR_BACKTRACE_STRING ERROR_BACKTRACE_STRING_START(0)
```
Comment (n-hutton) [inline libs/core/include/core/logger.hpp:790]: scheduled for demolition - will replace with backward-cpp trace library

File: libs/storage/include/storage/random_access_stack.hpp [hunk 2]
```diff
@@ -52,7 +52,7 @@ namespace storage {
  * The header for the stack optionally allows arbitrary data to be stored, which can be useful to
  * the user
  */
-template <typename T, typename D = uint64_t, typename STREAM = std::fstream>
```
Comment (n-hutton) [inline libs/storage/include/storage/random_access_stack.hpp:55]: This, and the test are not related but are minor - reverse this dependency injection from emumba a while back (also deletes disabled tests)

File: libs/math/include/math/ml/loss_functions/cross_entropy.hpp [hunk 3]
```diff
@@ -41,14 +41,14 @@ typename ArrayType::Type CrossEntropyLoss(
     ArrayType const &x, ArrayType const &y,
     typename ArrayType::SizeType n_classes = typename ArrayType::SizeType(2))
 {
-  using DataType = typename ArrayType::Type;
-  using SizeType = typename ArrayType::SizeType;
+  using DataType     = typename ArrayType::Type;
+  using SizeTypeHere = typename ArrayType::SizeType;
```
Comment (baykaner) [inline libs/math/include/math/ml/loss_functions/cross_entropy.hpp]: ?? why have you made this change>?

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]
Looks like a load of random binaries added to this, probably should sort out before merging

File: libs/ml/include/ml/layers/skip_gram.hpp [hunk 4]
```diff
@@ -93,9 +93,9 @@ class SkipGram : public Layer<T>
     {
       ArrayType slice_input   = inputs.front().get().Slice(b).Copy();
       ArrayType slice_context = inputs.back().get().Slice(b).Copy();
-      ArrayType output(ComputeOutputShape({slice_input, slice_context}));
-      this->Forward({slice_input, slice_context}, output);
-      results.push_back(output);
+      ArrayType output_here(ComputeOutputShape({slice_input, slice_context}));
```
Comment (baykaner) [inline libs/ml/include/ml/layers/skip_gram.hpp:96]: terrible name but sure that's fine I guess since this function needs reworking anyway

**Review by baykaner — APPROVED** [review]
looks good for the parts of the code I own

**Pull request closed without merging**